sync/clients/sync.py

```diff
@@ -194,6 +235,32 @@ def onboard_workflow(self, workspace_id, job_id: str, project_id: str) -> dict:
             )
         )
 
+    def get_latest_project_config_recommendation(self, project_id: str) -> ProjectConfiguration:
+        latest_recommendation = self.get_latest_project_recommendation(project_id)
+        if latest_recommendation.get("result"):
+            return latest_recommendation["result"][0]["recommendation"]["configuration"]
+        else:
+            return {"error": f"No project recommendation found for Project {project_id}"}
+
+    def get_cluster_definition_and_recommendation(
+        self, project_id: str, cluster_spec_str: str
+    ) -> dict:
+        cluster_recommendation = self.get_latest_project_config_recommendation(project_id)
+        return {
+            "cluster_recommendations": cluster_recommendation,
+            "cluster_definition": json.loads(cluster_spec_str),
+        }
+
+    def get_updated_cluster_defintion(
```

Comment:
Does this perform the `diff`?

Reply:
This produces an updated cluster definition with the recommendations merged in. The "diff" is in the method above which displays both the original cluster definition as well as the recommendations. Kartik and I decided it's best to provide the user with all of the information they need, and to do the diff-ing on their end